packages/components/dialog/src/DialogContent.tsx

```diff
@@ -13,10 +21,26 @@ export interface ContentProps extends RadixDialog.DialogContentProps, DialogCont
 
 export const Content = forwardRef(
   (
-    { children, className, isNarrow = false, size = 'md', ...rest }: ContentProps,
+    { children: childrenProp, className, isNarrow = false, size = 'md', ...rest }: ContentProps,
     ref: Ref<HTMLDivElement>
   ): ReactElement => {
-    const { setIsFullScreen } = useDialog()
+    const getElementId = (element?: ReactElement) => {
+      return element ? (element.type as FC & { id?: string }).id : ''
+    }
+
+    const findElement = (...values: string[]) => {
+      const children = Children.toArray(childrenProp).filter(isValidElement)
+
+      return children.find(child => values.includes(getElementId(child) || ''))
+    }
+
+    const closeButton = findElement('CloseButton')
+
+    const { setIsFullScreen, setHasCloseButton } = useDialog()
```

Comment:
I think all the added logic here should be inside `useDialog` instead (the purpose of this hook was to centralize the dialog logic)

Reply:
Done. I had to install a package to be able to recursively map through the children. Otherwise, the `CloseButton` couldn't be found

Reply:
Using the `react-children-utilities` causes error in our CI, so instead I added the `deepFind` util as new **internal-utils**, see #1965